lib/date.js

```diff
@@ -36,34 +45,56 @@ Hoek.inherits(internals.Date, Any);
 
 internals.Date.prototype._base = function (value, state, options) {
 
+    if (this._unit) {
+        internals.isValidUnit(this._unit);
```

Comment:
Can't crash _base like that. I'd prefer you override .unit() and call any's implementation rather than that late check.